google-cloud-datastore/src/main/java/com/google/cloud/datastore/DatastoreImpl.java

```diff
@@ -400,6 +401,41 @@ protected Entity computeNext() {
     }
   }
 
+  @Override
+  public List<Key> reserveIds(Key... keys) {
+    ReserveIdsRequest.Builder requestPb = ReserveIdsRequest.newBuilder();
+    for (Key key : keys) {
+      IncompleteKey incompleteKey = IncompleteKey.fromPb(key.toPb());
+      requestPb.addKeys(incompleteKey.toPb());
```

Comment:
I don't think the conversion to an `IncompleteKey` here isn't necessary and can be simplified
```suggestion
      requestPb.addKeys(key.toPb());
```

Both `Key` and `IncompleteKey` use the same `toPb` on `BaseKey`.

Reply:
done